crates/typst-ide/src/definition.rs

```diff
@@ -0,0 +1,240 @@
+use ecow::{eco_vec, EcoString};
+use typst::foundations::{Label, Module, Selector, Value};
+use typst::model::Document;
+use typst::syntax::ast::AstNode;
+use typst::syntax::{ast, FileId, LinkedNode, Side, Source, Span, SyntaxKind};
+use typst::World;
+
+use crate::analyze::analyze_import;
+use crate::{deref_target, named_items, DerefTarget, NamedItem};
+
+/// Find the definition of item under the cursor.
+///
+/// Passing a `document` (from a previous compilation) is optional, but enhances
```

Comment:
copy-pasta from autocomplete

Reply:
I actually copied it from the documentation of `tooltip`.

Reply:
As discussed, we should add an overall README.md to describe all about these arguments together.